### PR TITLE
Feat/rss ecomm 3 11 display product price

### LINKED ISSUE #115: Issue RSS-ECOMM-3_11: Display Product Price and Sale Price (15 points)
## 📋 Description
The application should fetch and display the price 💲 for each product from the chosen API on the Detailed Product page. If the product is on sale, both the original price and the sale price should be displayed. The sale price should be clearly distinguished as the current price of the product.

## 🔨 Implementation Details
**Fetch Product Price:** Fetch the price details of the product from the chosen API. This might involve using the GET /{projectKey}/products/{ID} or GET /{projectKey}/products/key={key} endpoints for the commercetools API, which should include product price information.
**Display Price:** Display the price of the product on the Detailed Product page. Make sure the price is formatted correctly and easy to read.
**Display Sale Price:** If the product is on sale, fetch the sale price from the API and display it alongside the original price. Make sure it's clear which price is the current price and which price is the original price before the sale.
🎨 Visual Implementation Ideas
**Sale Price Highlight:** When a product is on sale, the sale price could be highlighted in a bright color 🌈 to draw attention. The original price might be crossed out next to it to make clear that it's no longer the current price.
**Price Formatting:** Both prices should be displayed with the appropriate currency symbol and two decimal places, regardless of whether the price ends in .00.
## ✅ Acceptance Criteria
- [x] The application successfully fetches and displays the price for each product from the chosen API.
- [x] If the product is on sale, both the original price and the sale price are displayed. The sale price is clearly distinguished as the current price of the product.
🔗 Useful Links and Resources
[Commercetools API endpoint for fetching product by ID](https://docs.commercetools.com/api/projects/products#get-product-by-id)
[Formatting currency in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat)
[Commercetools API endpoint for fetching product by ID](https://docs.commercetools.com/api/projects/products#get-product-by-id)

### PR DESCRIPTION
**Pull request by Tetiana-KET**
#### RSS-ECOMM-3_11: Display Product Price and Sale Price
close #115 

#### 🤔 This is a ...

- [x] 🌟 New task

#### Description
 The application successfully fetches and displays the price for each product from the chosen API.
 If the product is on sale, both the original price and the sale price are displayed. The sale price is clearly distinguished as the current price of the product.

#### Screenshots

image
image

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [x] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.
- [x] 👌 Have at least 1 approve.
- [x] 🌳 Branch from `develop`/`release/...` and target to `develop`/`release/...`.

### CODE REVIEW

File: src/pages/Product/utils/generateProductObj.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { ProductData } from '@commercetools/platform-sdk';
+
+type Attributes = {
```
Comment (Gromov-Roman) [inline src/pages/Product/utils/generateProductObj.ts]: Please make strict type
```js
interface ProductDetail {
  lengthInfo: string;
  ...
}
```

File: src/pages/Product/utils/generateProductObj.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  [key: string]: string;
+};
+const CENTS = 100;
```
Comment (Gromov-Roman) [inline src/pages/Product/utils/generateProductObj.ts]: Please remove and use priceAmount const

File: src/pages/Product/utils/generateProductObj.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    data.masterVariant.attributes.forEach((attribute) => {
+      productInfo[attribute.name] = attribute.value;
+    });
+    productInfo.lengthInfo = productInfo.length ? `Length: ${productInfo.length} cm` : '';
+    productInfo.widthInfo = productInfo.width ? `Width: ${productInfo.width} cm` : '';
+    productInfo.heightInfo = productInfo.height ? `Height: ${productInfo.height} cm` : '';
+    productInfo.volumeInfo = productInfo.volume ? `Volume: ${productInfo.volume} lt` : '';
+  }
+
+  if (data && data.masterVariant.prices) {
```
Comment (Gromov-Roman) [inline src/pages/Product/utils/generateProductObj.ts]: if (data?.masterVariant.prices) {

File: src/pages/Product/utils/generateProductObj.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+type Attributes = {
+  [key: string]: string;
+};
+const CENTS = 100;
+
+export function generateProductObj(data: ProductData | undefined): Attributes {
+  const productInfo: Attributes = {};
+
+  if (data?.masterVariant?.attributes) {
+    data.masterVariant.attributes.forEach((attribute) => {
```
Comment (Gromov-Roman) [inline src/pages/Product/utils/generateProductObj.ts:8]:
```js
const {length, width, height, volume} = data.masterVariant.attributes;
```

File: src/pages/Product/Product.page.tsx [hunk 5]
```diff
@@ -6,24 +6,24 @@ import { isErrorWithBody } from '@core/errorHandlers/ErrorWithBody';
...
 
 function ProductPage() {
   const { productKey = '' } = useParams<'productKey'>();
   const showMessage = useShowMessage();
   const { data } = useGetProduct(productKey, {
     onError: (e) => {
       if (isErrorWithBody(e) && isResourceNotFoundError(e.body.errors[0])) {
-        // TODO: createErrorMessage
-        showMessage('The Product not found', 'error');
+        showMessage(createAppErrorMessage(e), 'error');
```
Comment (JsPowWow) [inline src/pages/Product/Product.page.tsx]: The code above `if (isErrorWithBody(e) && isResourceNotFoundError(e.body.errors[0]))..`
is not required here, isn't?

File: src/pages/Product/utils/generateProductObj.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    productInfo.widthInfo = productInfo.width ? `Width: ${productInfo.width} cm` : '';
+    productInfo.heightInfo = productInfo.height ? `Height: ${productInfo.height} cm` : '';
+    productInfo.volumeInfo = productInfo.volume ? `Volume: ${productInfo.volume} lt` : '';
+  }
+
+  if (data && data.masterVariant.prices) {
+    if (data.masterVariant.prices[0].value.centAmount) {
+      productInfo.basePrice = `${data.masterVariant.prices[0].value.centAmount / CENTS} €`;
+    }
+    if (data.masterVariant.prices[0].discounted && data.masterVariant.prices[0].discounted.value.centAmount) {
```
Comment (Gromov-Roman) [inline src/pages/Product/utils/generateProductObj.ts]: data.masterVariant.prices[0].discounted?.value.centAmount

File: src/pages/Product/utils/generateProductObj.ts (hunk 2 above)
Comment (Tetiana-KET) [reply src/pages/Product/utils/generateProductObj.ts]: done

File: src/pages/Product/utils/generateProductObj.ts (hunk 3 above)
Comment (Tetiana-KET) [reply src/pages/Product/utils/generateProductObj.ts]: done

File: src/pages/Product/utils/generateProductObj.ts (hunk 4 above)
Comment (Tetiana-KET) [reply src/pages/Product/utils/generateProductObj.ts:8]: I separately take out the value of the name field to make it a key, and the value of the value field. I don't understand how to change it

File: src/pages/Product/Product.page.tsx (hunk 5 above)
Comment (Tetiana-KET) [reply src/pages/Product/Product.page.tsx]: Done, thank you

File: src/pages/Product/utils/generateProductObj.ts (hunk 6 above)
Comment (Tetiana-KET) [reply src/pages/Product/utils/generateProductObj.ts]: done

**Review by Gromov-Roman — APPROVED** [review]

**Pull request merged**